src/chain.js

```diff
@@ -11,19 +11,25 @@ var map = require('./map');
  *
  * Dispatches to the `chain` method of the second argument, if present,
  * according to the [FantasyLand Chain spec](https://github.com/fantasyland/fantasy-land#chain).
+ * Treats the second argument as a FantasyLand Chain, when the the second argument is a function
+ * of the form `a -> x -> b`.
  *
  * @func
  * @memberOf R
  * @since v0.3.0
  * @category List
  * @sig Chain m => (a -> m b) -> m a -> m b
+ * @sig (x -> a) -> (a -> x -> b) -> (x -> b)
```

Comment:
No one seems to be mentioning that the [fantasy-land spec][fl] signature is different from ours.  I think that's much of the problem.  In FL, the signature is

```hs
chain :: Chain m => m a ~> (a -> m b) -> m b
```

and that's what our function-branch supports.  I haven't checked yet about our list-branch, as this stuff makes my head hurt.  All this started with an answer from @bergus in a [StackOverflow question][so].  But that answer seemed to be speculative, just assuming that Ramda would work as expected for functions, which, it turns out, it does.  I think the problem is not that the documentation is incomplete, but that it's entirely wrong, at least for functions.  And if it's wrong for functions, and right for other types, we've got significant problems.



  [fl]: https://github.com/fantasyland/fantasy-land#chain
  [so]: http://stackoverflow.com/questions/40026018